src/components/EditorCardsDrawer/EditorCardItem.tsx

```diff
@@ -29,145 +43,196 @@ export const EditorCardItem = ({
+    onClose: onDeleteWarningModalClose,
+  } = useDisclosure()
   const { errors } = methods.formState
 
   return (
-    <Editable.DraggableAccordionItem
-      index={index}
-      title={methods.watch(`cards.${index}.title`) || `Card #${index + 1}`}
-      draggableId={`cards-${index}`}
-      isInvalid={_.some(errors.cards?.[index])}
-      isNested
-    >
-      <VStack w="100%" spacing="1rem" p="0.5rem">
-        {/* Card image */}
-        {methods.watch("isDisplayImage") && (
-          <>
-            <Box w="100%">
-              <FormContext
-                hasError={!!errors.cards?.[index]?.image}
-                onFieldChange={(e) =>
-                  methods.setValue(`cards.${index}.image`, e.target.value)
-                }
+    <>
+      <Modal
+        isOpen={isDeleteWarningModalOpen}
+        onClose={onDeleteWarningModalClose}
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalCloseButton />
+
+          <ModalHeader>
+            <Text as="h4" textStyle="h4">
+              Are you sure you want to delete this card?
+            </Text>
+          </ModalHeader>
+
+          <ModalBody>
+            <Text as="p" textStyle="body-2" mt="0.5rem">
+              This cannot be undone.
+            </Text>
+          </ModalBody>
+
+          <ModalFooter>
+            <HStack w="100%" spacing={4} justifyContent="flex-end" mt="0.5rem">
+              <Button
+                variant="clear"
+                colorScheme="neutral"
+                onClick={onDeleteWarningModalClose}
+              >
+                Cancel
+              </Button>
+              <Button
+                colorScheme="critical"
+                onClick={() => {
+                  onDelete(`cards-${index}`, "cards")
+                }}
               >
-                <FormLabel mt="0.25rem" mb="0.5rem" isRequired>
-                  Image
-                </FormLabel>
-                <FormFieldMedia
-                  id={`cards.${index}.image`}
-                  placeholder="Browse an image"
-                  register={methods.register}
-                  inlineButtonText="Browse"
+                Delete
+              </Button>
+            </HStack>
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
+
+      <Editable.DraggableAccordionItem
+        index={index}
+        title={methods.watch(`cards.${index}.title`) || `Card #${index + 1}`}
+        draggableId={`cards-${index}`}
+        isInvalid={_.some(errors.cards?.[index])}
+        isNested
+      >
+        <VStack w="100%" spacing="1rem" p="0.5rem">
+          {/* Card image */}
+          {methods.watch("isDisplayImage") && (
+            <>
+              <Box w="100%">
+                <FormContext
+                  hasError={!!errors.cards?.[index]?.image}
+                  onFieldChange={(e) => {
+                    methods.setValue(`cards.${index}.image`, e.target.value)
+                    methods.trigger()
```

Comment:
could i get some context on why we are manually trigger validation here? my understanding is that if our mode is `onTouch`, this shouldn't be required?

Reply:
Hmm cos we are setting the values using `setValue`, which for some reason doesn't trigger a validation.